### PR TITLE
Workaround to allow autocompletion labels containing spaces

### PR DESCRIPTION
**Pull request by rayanamal**

### CODE REVIEW

File: init.lua [hunk 1]
```diff
@@ -844,7 +844,12 @@ textadept.editing.autocompleters.lsp = function()
 			label = symbol.filterText or symbol.label
 			snippets[label] = symbol.insertText
 		end
-		-- TODO: some labels can have spaces and need proper handling.
+		
+		-- If a label contains spaces, use '|' as the separator. TODO: This is not ideal. A label might contain '|' too.
```
Comment (orbitalquark) [inline init.lua:848]: Yes, you're right. I still need to sit and think on this. When I'm testing, I often type `buffer:auto_c_show(0, 'foo bar baz')` into the command entry to popup an autocompletion list, and this inherently relies on the default value of ' '. I should probably rewire my thinking and use `auto_c_show(0,table.concat({'foo','bar','baz'},string.byte(auto_c_separator)))`, but that's a lot of typing :)

Still, thanks for bringing this up and I will put it on my TODO list.

Comment (rayanamal) [reply init.lua:848]: The solution I currently settled in is to patch lsp module for it to use textredux for filename autocompletion functionality. (which was the use case requiring proper handling of spaces). 

Personally I see this workaround as a good enough fix. Even if not ideal.

I didn’t close the PR, it automatically got closed when I deleted my fork btw. Can’t reopen the PR. Reopen and merge if you agree.

**Comment by orbitalquark** [comment]
Note to self: the documentation for `textadept.editing.autocompleters` says that spaces are used by default, so autocompleter functions may not be written with using `buffer.auto_c_separator` in mind.

**Comment by orbitalquark** [comment]
I've committed what I think is proper fix here: https://github.com/orbitalquark/textadept-lsp/commit/78e72c28acdd6d693cdc1c547f3803769a632d15

**Pull request closed without merging**